packages/grid/_modules_/grid/hooks/features/treeData/useGridTreeData.tsx

```diff
@@ -51,9 +51,20 @@ export const useGridTreeData = (
       colDefOverride = propGroupingColDef;
     }
 
+    const { hideDescendantCount, ...colDefOverrideProperties } = colDefOverride ?? {};
+
+    const commonProperties: Omit<GridColDef, 'field' | 'editable'> = {
+      ...GRID_TREE_DATA_GROUP_COL_DEF,
+      renderCell: (params) => (
```

Comment:
Why do you not `hideDescendantCount` to the `colDef` and use this parameter in the `<GridTreeDataGroupingCell />` component like `colDef.valueOptions` for `<GridEditSingleSelectCell />`?

Reply:
On which `colDef` do you want to add it ?

Reply:
I wanted to avoid adding properties on the `colDef` that are only useful for the grouping column.
But if the team feels like it's better to pass it to the `colDef` then I just have to move `hideDescendantCount` from `GridGroupingColDefOverride` to `GridColDef` and access it through the `colDef`prop in the cell component.

Reply:
Make sense, but when defining `groupingColDef` developers will probably expect their props to be in the `colDef`